api/src/test/java/org/apache/iceberg/TestMetricsSerialization.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestMetricsSerialization {
+
+  @Test
+  public void testSerialization() throws IOException, ClassNotFoundException {
+    Metrics original = generateMetrics();
+
+    byte[] serialized = serialize(original);
+    Metrics result = deserialize(serialized);
+
+    assertEquals(original, result);
+  }
+
+  @Test
+  public void testSerializationWithNulls() throws IOException, ClassNotFoundException {
+    Metrics original = generateMetricsWithNulls();
+
+    byte[] serialized = serialize(original);
+    Metrics result = deserialize(serialized);
+
+    assertEquals(original, result);
+  }
+
+  private static byte[] serialize(Metrics metrics) throws IOException {
+    try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+      ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+      objectOutputStream.writeObject(metrics);
```

Comment:
Don't we need to flush the object output stream?

Reply:
Good catch! Thanks!
Done